.github/scripts/check_symbols.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: 2024 Second State INC
+
+# Script to check that libwasmedge doesn't expose unnecessary symbols
+# This script compares the exported symbols against a whitelist to ensure
+# only intended API symbols are exposed.
+
+set -euo pipefail  # Exit on error, undefined vars, pipe failures
+
+# Default paths
+SYMBOLS_PATH=${SYMBOLS_PATH:-"$(pwd)"}
+WHITELIST_FILE="$SYMBOLS_PATH/lib/api/whitelist.symbols"
+
+# Detect OS and set library path accordingly
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    LIB_PATH="$SYMBOLS_PATH/build/lib/api/libwasmedge.so"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    LIB_PATH="$SYMBOLS_PATH/build/lib/api/libwasmedge.dylib"
+elif [[ "$OSTYPE" == "msys" ]] || [[ "$OSTYPE" == "cygwin" ]]; then
+    LIB_PATH="$SYMBOLS_PATH/build/lib/api/wasmedge.dll"
+else
+    echo "Unsupported OS: $OSTYPE"
+    exit 1
+fi
+
+echo "Checking symbol exposure for libwasmedge..."
+echo "Library path: $LIB_PATH"
+echo "Whitelist file: $WHITELIST_FILE"
+
+# Check if library exists
+if [ ! -f "$LIB_PATH" ]; then
+    echo "Error: Library not found at $LIB_PATH"
+    echo "Make sure you have built WasmEdge first with: cmake --build build"
```

Comment:
I don't think this error message will be helpful. The reason is that people may have their own locations for the `LIB_PATH`; it is not related to the building process.